inference-engine/src/inference_engine/include/openvino/runtime/gpu/dx.hpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief a header that defines wrappers for internal GPU plugin-specific
+ * shared Video Acceleration device contexts
+ * and shared memory blobs which contain Video Acceleration surfaces
+ *
+ * @file openvino/runtime/gpu/dx.hpp
+ */
+#pragma once
+
+#include <d3d11.h>
+
+#include <memory>
+#include <string>
+
+#include "openvin/runtime/gpu/ocl.hpp"
+
+namespace ov {
+namespace runtime {
+namespace gpu {
+
+/**
+ * @brief This class represents an abstraction for GPU plugin remote tensor
+ * which is shared with Direct3D 11 buffer.
+ * The plugin object derived from this class can be obtained with create_tensor() call.
+ * @note User can also obtain OpenCL buffer handle from this class.
+ */
+class D3DBufferTensor : public ClBufferTensor {
+    using ClBufferTensor::ClBufferTensor;
+
+public:
+    /**
+     * @brief Checks that type defined runtime paramters are presented in remote object
+     * @param remote_tensor remote tensor to check
+     */
+    static void type_check(const RemoteTensor& remote_tensor) {
+        remote_type_check(
+            remote_context.get_params(),
+            {{GPU_PARAM_KEY(DEV_OBJECT_HANDLE), {}}, {GPU_PARAM_KEY(SHARED_MEM_TYPE), {GPU_PARAM_VALUE(DX_BUFFER)}}});
+    }
```

Comment:
@ilyachur @apankratovantonp can we use ngraph RTTI here:
```
// define in the class
OPENVINO_OP("D3DBufferTensor ", "", ClBufferTensor );
```
And in `RemoteTensor::is`:
```c++
template <typename T>
bool is() {
    static_assert(std::is_base_of<RemoteContext, T>::value,
                      "Could not check type that dose not inherits from RemoteContext");
    return ov::is_type<T>(*this);
}
```